src/node_nanomsg.cc

```diff
@@ -56,10 +56,16 @@ NAN_METHOD(Chan) {
   int s = Nan::To<int>(info[0]).FromJust();
   int level = NN_SUB;
   int option = Nan::To<int>(info[1]).FromJust();
-  Nan::Utf8String str(info[2]);
 
-  info.GetReturnValue().Set(Nan::New<v8::Number>(
+  if (node::Buffer::HasInstance(info[2])) {
+    info.GetReturnValue().Set(Nan::New<v8::Number>(nn_setsockopt(
+        s, level, option, node::Buffer::Data(info[2]),
+        node::Buffer::Length(info[2]))));
+  } else {
+    Nan::Utf8String str(info[2]);
+    info.GetReturnValue().Set(Nan::New<v8::Number>(
         nn_setsockopt(s, level, option, *str, str.length())));
+    }
```

Comment:
There's a fair amount of duplication on either side of that condition.  Perhaps you can call `nn_setsockopt` two different ways in either branch, but then have the common tail construct the v8::Number and call Set after?

Reply:
I used the same code style as in Send method but sure!